### ISSUE TITLE
document primary support for node 4 and 5 in README

### ISSUE THREAD

**Issue by ralphtheninja** [ralphtheninja/pgdown#10]
afaik 0.12 will work with `--harmony`, correct @deanlandolt ?
what about 0.10? can this be supported by using `babelify`?

**Comment by deanlandolt** [comment]
IIRC the only two es features we're using are arrow fns and template strings (pretty useful for authoring sql strings -- and even more useful when we start using something like [`pg-template-tag`](https://github.com/XeCycle/pg-template-tag) for proper escapement). This is all very babelable -- both are just straight transpiles, no shimming cruft necessary.

**Comment by ralphtheninja** [comment]
> This is all very babelable -- both are just straight transpiles, no shimming cruft necessary.

:+1: Cool. So it's basically a none issue.

**Comment by deanlandolt** [comment]
Might be nice to add some node versions to travis, and maybe try babelify as a dev-dep for older nodes.

**Comment by ralphtheninja** [comment]
> Might be nice to add some node versions to travis, and maybe try babelify as a dev-dep for older nodes.

Good call. On it.

**Comment by ralphtheninja** [comment]
@deanlandolt Using `babelify` requires `browserify` right? Or is there some other way to babelify the code without using browserify?

**Comment by juliangruber** [comment]
babelify is just the browserify transform, that's correct. for regular builds, you can use babel-cli:

``` js
"scripts": {
  "build": "babel src -d lib"
}
```

see https://babeljs.io/docs/setup/#babel_cli

**Comment by ralphtheninja** [comment]
:+1:

**Comment by deanlandolt** [comment]
IIRC there's also a babel require hook, so we can just conditionally require this on older nodes to avoid any explicit build step.

**Comment by ralphtheninja** [comment]
This is a mess imo, or perhaps I just don't get it, or _both_ ;)

If I try to require `babel-register` in e.g. `test/index.js` like this (using node 0.12 to test):

``` js
require('babel-register')
require('./pgdown')            
require('./encoding')          
require('./abstract-leveldown')
```

I get the following error when running `npm test`:

![babelerror1](https://cloud.githubusercontent.com/assets/308049/13642221/e7c58214-e61b-11e5-9b5b-f708cd7ecb94.png)

And I get the same error if I do:

```
$ babel test -d test-babel
$ PGDATABASE=${PGDATABASE:-postgres} PGUSER=${PGUSER:-postgres} node test-babel/
```

**Comment by ralphtheninja** [comment]
I can get around _that_ by doing `node --harmony` but then I instead get problems with template strings:

![babelerror2](https://cloud.githubusercontent.com/assets/308049/13642443/d091a2d4-e61c-11e5-8ea5-2eb1ca2f397f.png)

**Comment by ralphtheninja** [comment]
Oh I might need to specify some presets maybe http://babeljs.io/docs/plugins/preset-es2015/

**Comment by ralphtheninja** [comment]
> IIRC there's also a babel require hook, so we can just conditionally require this on older nodes to avoid any explicit build step.

I guess this means `babel-register` and `babel-preset-es2015` would have to be _dependencies_, right?

**Comment by juliangruber** [comment]
yeah i think we need the es2015 preset. but we should only add them as devdeps, the require hook shouldn't be published

**Comment by deanlandolt** [comment]
Exactly -- just switch on node version in the test/index.js and load the appropriate require hook. I can easily kill all the template strings if that's giving you grief. How far back can we go w/ node versions using `--harmony` to get `arrow functions` and `const`? If `0.12`, that seems good enough. Otherwise, we'd need to babel this shit up anyway, so just do the `babel-register`, `babel-preset-es2015` dance.

**Comment by ralphtheninja** [comment]
> yeah i think we need the es2015 preset. but we should only add them as devdeps, the require hook shouldn't be published

Cool. I'll fix up a PR for running the tests on 0.10 and 0.12 then.
